query/raw_result.go

```diff
@@ -0,0 +1,210 @@
+package query
+
+import (
+	"github.com/xichen2020/eventdb/document/field"
+)
+
+const (
+	defaultRawResultsCapacity = 4096
+)
+
+// RawResult is a single raw result returned from a raw query.
+// TODO(xichen): Implement `MarshalJSON` to only marshal the `Data` field without the `data tag.
+type RawResult struct {
+	Data string
+
+	// Fields for joining and sorting purposes. These fields are empty for unsorted raw results.
+	DocID         int32
+	OrderByValues []field.ValueUnion
+	OrderIdx      int
```

Comment:
So this is the index of the raw result when the raw results are ordered as dictated by the query (e.g., if there are two raw results sorted by time in descending order, then the first raw result has an `OrderIdx` of 0, and the second one has an `OrderIdx` of 1). This is so that if the raw results are sorted in a different order (e.g., by doc ID), we can resort the results later on to restore the original order.